File: 프로그래머스/1/12930. 이상한 문자 만들기/이상한 문자 만들기.java
class Solution {
    public String solution(String s) {
        char[] charArr = s.toCharArray();
        
        StringBuilder sb = new StringBuilder();
        int j = 0;
        
        for (int i = 0; i < charArr.length; i++) {
            char c = charArr[i];
            
            if(c == ' ') {
                sb.append(c);
                j = 0;
                continue;
            }
            
            if(j % 2 == 0) {
                sb.append(String.valueOf(Character.toUpperCase(c)));                
            } else {
                sb.append(String.valueOf(Character.toLowerCase(c)));
            }
            
            j++;
        }
        
        return sb.toString();
    }
}

File: 프로그래머스/1/12930. 이상한 문자 만들기/이상한 문자 만들기.js
function solution(s) {
    let temp = s.split(" ");
    let answer = [];
    
    for(let i = 0; i < temp.length; i++) {
        let str = '';
        
        for(let j = 0; j < temp[i].length; j++) {
            if(j % 2 == 0) {
                str += temp[i][j].toUpperCase();
            } else {
                str += temp[i][j].toLowerCase();
            }
        }
        
        answer.push(str);
    }
    
    return answer.join(" ");
}